feat(fps): support x-nv-video[0].clientRefreshRateX100 for requesting fractional NTSC framerates

andygrundman opened the pull request:
## Description
This PR brings back an old GFE parameter for specifying fractional refresh rates. If the param is specified, Sunshine will try to deliver the exact rate requested, which can be helpful for clients using Apple TV, Xbox, and others that might be stuck at 59.94hz or 119.88hz.

Common NTSC rates are hardcoded to their exact values, because 59.94 is not really the true rate, it's defined as 60000/1001. Just for fun, 23.976hz film is also supported if either 2397 or 2398 is supplied.

### Screenshot
```
[2025-06-27 22:08:45.366]: Info: Display refresh rate [119.999Hz]
[2025-06-27 22:08:45.366]: Info: Requested frame rate [60000/1001 exactly 59.94 fps]
```
## Type of Change
- [x] New feature (non-breaking change which adds functionality)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

ReenigneArcher commented on tools/CMakeLists.txt:
The FFmpeg stuff has to be added here because of the changes to `utility.h`? Would it make more sense to put the `framerateX100_to_rational` function somewhere else that won't affect the dxgi tool?

andygrundman replied:
Yeah that was actually the most annoying part, getting everything to compile after adding an ffmpeg data type and function. It made sense as a misc. utility function to throw it in utility, but I'm happy to move it if you have a better place for it to go.

andygrundman commented:
I moved the function to video, and reverted the stuff in utility.h affecting tools. No need to include rational.h now either. The only weird thing left is in `src/platform/windows/display_base.cpp` where I needed to do this due to video namespace issues.

```
AVRational fps = ::video::framerateX100_to_rational(config.framerateX100);
```

ns6089 commented:
Alternatively instead of x100 workaround you can introduce new fractional SDP parameter (or two). This should remove intermediate steps and extend the usability, like creating new virtual display modes in runtime.

ns6089 commented:
> Are you trolling

I'm not. You're introducing technical debt into client-server contract by adding support for poorly designed GFE attribute. If clients desires particular precision, it should request particular precision without intermediate guesswork.

ns6089 commented:
Either way [here](https://github.com/moonlight-stream/moonlight-common-c/pull/113) are the related changes to `moonlight-common-c`, so if @cgutman greenlights them, you won't need to write any extra code.

mcourteaux commented:
This PR broke several hardcoded config initializer expressions, because of the newly-inserted field in the config.

https://github.com/LizardByte/Sunshine/blob/eb72930a8ef40b19843e92e745c2bfcdc13baea1/src/video.cpp#L2469-L2471

https://github.com/LizardByte/Sunshine/blob/eb72930a8ef40b19843e92e745c2bfcdc13baea1/src/video.cpp#L2566

https://github.com/LizardByte/Sunshine/blob/eb72930a8ef40b19843e92e745c2bfcdc13baea1/src/video.cpp#L2573

I'll PR a fix today.
